tests/bazel_run_tests.sh

```diff
@@ -15,6 +15,19 @@ function test_duplicate_version_warning() {
   expect_log "Successfully pinned resolved artifacts"
 }
 
+function test_m2local_testing_found_local_artifact() {
```

Comment:
can you also add a couple of more tests:

1) do a full build on the jar target (`@m2local_testing//:com_example_kt`) instead of running the `pin` executable
2) manually create the m2local directory instead of relying on java_export, so we have a test that doesn't depend on java_export's semantics

Reply:
Added 1 test using the `@m2local_testing//:com_example_kt` build target.

Also added another test, that builds the kt library, and copies binaries from bazel-bin and populate it into m2local, and then build the testing target `@m2local_testing//:com_example_kt)` (after a bazel clean) to validate that it is fetched from m2local. Checksums don't exist in this testcase, so I created a WORKSPACE repository that has `fail_on_missing_checksums = False`.